utilities/list-methods.metta

```diff
@@ -0,0 +1,71 @@
+
+
+(: List (-> $a Type))
+(: Nil (List $a))
+(: Cons (-> $a (List $a) (List $a)))
+
+
+;; Define List.foldr
+;; Fold a tuple from right to left
+(: List.foldr (-> (-> $a $b $b) $b (List $a) $b))
+(= (List.foldr $f $i Nil) $i)
+(= (List.foldr $f $i (Cons $h $t)) ($f $h (List.foldr $f $i $t)))
+;; ! (List.foldr * 1 (Cons 1 (Cons 2 (Cons 3 (Cons 4 (Cons 5 Nil))))))
+
+;; Define List.sum
+(: List.sum (-> (List Number) Number))
+(= (List.sum $xs) (List.foldr + 0 $xs))
+;; ! (List.sum (Cons 1 (Cons 2 (Cons 3 (Cons 2 (Cons 3 Nil)))))) ;; [11]
+
+
+
+; convert (a b c) to (Cons a (Cons b (Cons c Nil)))-> Type definition need modification 
```

Comment:
Remove this function as it is not stable yet. Maybe put it on the experimental branch until we decide how to exactly implement it.